### DISCUSSION TITLE
false audio bit rate

### DISCUSSION THREAD

**Discussion by achnoe** [aclap-dev/video-downloadhelper#321] [Help and Q&A]
### Your question

When I download videos from youtube with the option Audio-only download then in the header of the mp3 file is a false bit rate declined. Corresponding to the used bit rate of 128 kbit/s the byte at the file position 0x2F should be 0x90 but is 0x50. The rate description in the audio frames is right, therefore the files are played correctly. But in apps like MusicBee that uses the header the lenght of the audio track is listed false. I don't know whether the failure is already in the youtube stream or in the writing to the file.
![grafik](https://github.com/aclap-dev/video-downloadhelper/assets/164529582/43489703-956a-4edb-b914-7f5cb8f47c0c)

### Addon & CoApp version and details

version 8.2.2.8 stable
platform windows 11
VdhCoApp 2.0.19 installed

**Comment by achnoe** [comment]
With the latest beta version 9.3.7.0 this problem is still pending.

**Comment by Mattmjs** [comment]
What is the problem ? does the file not play in the musicbee app. If you can download another way you can verify or not if the fault
is still there. But I'd just use another media player and not worry about it.

**Comment by achnoe** [comment]
It 's not a great problem. The files are played right in the musicbee app, but the times in the title list are false because there is used the false bitrate value of the file header for time prediction. Other players seems to use another file parts with correct values. Maybe this byte isn't correct also in the youtube source file, but within streaming data the header isn't used and it doesn't care. I fix this byte by using a hex editor and all is good.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

You appear to know a whole lot more about this than I do.  I have noticed on occasion that VLC reports an incorrect duration for MP3 files.  Quite often it will claim the file is twice as long as it really is.  Maybe that's related to the issue you're dealing with.  It doesn't seem to be an issue that bothers VLC.  The file plays everything that is there for the correct duration.  It just never reports a correct total duration of the media file.

I suspect that what you're doing is what has to be done.  Given that YouTube is so non-standard about most things, I would suspect the error in the header is YouTube's fault.  I would recommend you continue to fix things as you are doing & get on with your life.

**Reply by achnoe** [reply]
Firstly, English is a foreign langage for me, excuse me language failures. Some years ago I wrote an app with mp3 content. From this time I have some knowledge yet.
MP3 files has many different faces. The pure internet mp3 stream consists of independant frames that contain all necessary information for playing. The stream is endless. By storing a limited stream, for example a music title, there is added at the beginning a header with general information. It contains date for bit rate, lenghts of a frame, number of frames and others. From this info many players calculate the duration. A false bit rate gives a false duration. I use a hex editor to fix the failure, in my case I use HxD for windows. In my first post you can see the first 48 (30hex) bytes of the header. It always starts with the code for "ID3". From this point the byte # 2Fhex contains among other things the identifier for the bit rate (3 bits). In Youtube downloads there is often given a wrong value, in my case the value 50h instead of the correct 90h. But if the file contains mp3 tags for title and so on this header is not the first part of the file. Than firstly you have to look for the header beginning "ID3" und then to count 2Fh forward. This procedure I do manually and it takes less than a minute for me. If this behaviour by youtube downloads continues I think about writing a script for fixing the failure. The problem is I dealed with programming microcontrolers but nor writing windows apps. Therefore it would take some time. If I nevertheless write an app I will report it at this place.
